python/huggingfaceserver/huggingfaceserver/encoder_model.py

```diff
@@ -125,12 +124,13 @@ def load(self) -> bool:
         model_id_or_path = self.model_id_or_path
 
         self.max_length = _get_and_verify_max_len(self.model_config, self.max_length)
+        model_cls = get_model_class_for_task(self.task)
 
         # device_map = "auto" enables model parallelism but all model architcture dont support it.
         # For pre-check we initialize the model class without weights to check the `_no_split_modules`
         # device_map = "auto" for models that support this else set to either cuda/cpu
         with init_empty_weights():
-            self._model = AutoModel.from_config(self.model_config)
+            self._model = model_cls.from_config(self.model_config)
```

Comment:
What's the difference between AutoModel and `get_model_class_for_task(self.task)` ?

Reply:
AutoModel loads a model without task specific head while AutoModelFor* loads model with task specific head.

Reply:
@yuzisun more info here https://stackoverflow.com/a/77588279 interesting